sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/ContainerRegistryClient.java

```diff
@@ -0,0 +1,109 @@
+     */
+    ContainerRegistryClient(ContainerRegistryAsyncClient asyncClient) {
+        this.asyncClient = asyncClient;
+    }
+
+    /**
+     * List repositories.
+     *
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws AcrErrorsException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return list of repositories.
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PagedIterable<String> listRepositories() {
+        return new PagedIterable<>(this.asyncClient.listRepositories());
+    }
+
+    /**
+     * List repositories.
+     *
+     * @param options for the list repositories call.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws com.azure.containers.containerregistry.models.AcrErrorsException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return list of repositories.
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PagedIterable<String> listRepositories(ListRepositoriesOptions options) {
+        return new PagedIterable<>(this.asyncClient.listRepositories(options));
+    }
+
+    /**
+     * List repositories.
+     *
+     * @param options The options to use while getting the list of repositories.
+     * @param context The context to associate with this operation.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws AcrErrorsException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return list of repositories.
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PagedIterable<String> listRepositories(ListRepositoriesOptions options, Context context) {
+        Objects.requireNonNull(options);
+
+        Integer pageSize = options.getMaxPageSize();
+        return new PagedIterable<>(new PagedFlux<>(
+            () -> asyncClient.listRepositoriesSinglePageAsync(pageSize, context),
+            token -> asyncClient.listRepositoriesNextSinglePageAsync(token, context)));
+    }
+
+    /**
+     * Delete the repository identified by `name`.
+     *
+     * @param name Name of the image (including the namespace).
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws AcrErrorsException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return deleted repository.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public DeleteRepositoryResult deleteRepository(String name) {
+        return this.asyncClient.deleteRepository(name).block();
```

Comment:
Would call into the API below with `Context.NONE`.

Reply:
I thought we try to have minimum code in sync clients and simply delegate things to async clients.

---
In reply to: [602631431](https://github.com/Azure/azure-sdk-for-java/pull/20139#discussion_r602631431) [](ancestors = 602631431)

Reply:
I've seen both flows of having <simple sync overload> -> <simple async overload> and <simple sync overload> -> <complex sync overload>.